cfgmgr/vxlanmgrd.cpp

```diff
@@ -45,26 +47,71 @@ int main(int argc, char **argv)
     try
     {
 
-        DBConnector cfgDb("CONFIG_DB", 0);
-        DBConnector appDb("APPL_DB", 0);
-        DBConnector stateDb("STATE_DB", 0);
+        DBConnector cfgDb(CONFIG_DB, DBConnector::DEFAULT_UNIXSOCKET, 0);
+        DBConnector appDb(APPL_DB, DBConnector::DEFAULT_UNIXSOCKET, 0);
+        DBConnector stateDb(STATE_DB, DBConnector::DEFAULT_UNIXSOCKET, 0);
+
+        WarmStart::initialize("vxlanmgrd", "swss");
+        WarmStart::checkWarmStart("vxlanmgrd", "swss");
+        if (WarmStart::isWarmStart())
+        {
+            WarmStart::setWarmStartState("vxlanmgrd", WarmStart::INITIALIZED);
+        }
 
         vector<std::string> cfg_vnet_tables = {
             CFG_VNET_TABLE_NAME,
             CFG_VXLAN_TUNNEL_TABLE_NAME,
             CFG_VXLAN_TUNNEL_MAP_TABLE_NAME,
+            CFG_VXLAN_EVPN_NVO_TABLE_NAME,
         };
 
         VxlanMgr vxlanmgr(&cfgDb, &appDb, &stateDb, cfg_vnet_tables);
-
         std::vector<Orch *> cfgOrchList = {&vxlanmgr};
-
         swss::Select s;
         for (Orch *o : cfgOrchList)
         {
             s.addSelectables(o->getSelectables());
         }
 
+        /*
+         * swss service starts after interfaces-config.service which will have
+         * switch_mac set.
+         * Dynamic switch_mac update is not supported for now.
+         */
+        Table table(&cfgDb, "DEVICE_METADATA");
+        std::vector<FieldValueTuple> ovalues;
+        table.get("localhost", ovalues);
+        auto it = std::find_if( ovalues.begin(), ovalues.end(), [](const FieldValueTuple& t){ return t.first == "mac";} );
+        if ( it == ovalues.end() ) {
+            throw runtime_error("couldn't find MAC address of the device from config DB");
+        }
+        gMacAddress = MacAddress(it->second);
+
+        auto in_recon = true;
+        vxlanmgr.beginReconcile(true);
+
+        if (WarmStart::isWarmStart())
+        {
+            vxlanmgr.waitTillReadyToReconcile();
+            vxlanmgr.restoreVxlanNetDevices();
+            WarmStart::setWarmStartState("vxlanmgrd", WarmStart::REPLAYED);
```

Comment:
There is no `REPLAYED` state in the sonic-swss-common [PR339](https://github.com/Azure/sonic-swss-common/pull/339)

Reply:
Will add it in sonic-swss-common

Reply:
Separate PR on swss-common will be filed for new states added as part of warm reboot.